test/e2e/settings/custom_template.go

```diff
@@ -0,0 +1,752 @@
+					{{ end }}
+					{{ end }}
+					{{/* comment PEM sha is required to detect changes in the generated configuration and force a reload */}}
+					# PEM sha: {{ $server.SSLCert.PemSHA }}
+					ssl_certificate                         {{ $server.SSLCert.PemFileName }};
+					ssl_certificate_key                     {{ $server.SSLCert.PemFileName }};
+					{{ if not (empty $server.SSLCert.FullChainPemFileName)}}
+					ssl_trusted_certificate                 {{ $server.SSLCert.FullChainPemFileName }};
+					ssl_stapling                            on;
+					ssl_stapling_verify                     on;
+					{{ end }}
+					{{ end }}
+			
+					{{ if not (empty $server.AuthTLSError) }}
+					# {{ $server.AuthTLSError }}
+					return 403;
+					{{ else }}
+			
+					{{ if not (empty $server.CertificateAuth.CAFileName) }}
+					# PEM sha: {{ $server.CertificateAuth.PemSHA }}
+					ssl_client_certificate                  {{ $server.CertificateAuth.CAFileName }};
+					ssl_verify_client                       {{ $server.CertificateAuth.VerifyClient }};
+					ssl_verify_depth                        {{ $server.CertificateAuth.ValidationDepth }};
+					{{ if not (empty $server.CertificateAuth.ErrorPage)}}
+					error_page 495 496 = {{ $server.CertificateAuth.ErrorPage }};
+					{{ end }}
+					{{ end }}
+			
+					{{ if not (empty $server.SSLCiphers) }}
+					ssl_ciphers                             {{ $server.SSLCiphers }};
+					{{ end }}
+			
+					{{ if not (empty $server.ServerSnippet) }}
+					{{ $server.ServerSnippet }}
+					{{ end }}
+			
+					{{ range $location := $server.Locations }}
+					{{ $path := buildLocation $location }}
+					{{ $proxySetHeader := proxySetHeader $location }}
+					{{ $authPath := buildAuthLocation $location }}
+			
+					{{ if not (empty $location.Rewrite.AppRoot)}}
+					if ($uri = /) {
+						return 302 {{ $location.Rewrite.AppRoot }};
+					}
+					{{ end }}
+			
+					location {{ $path }} {
+					}
+					{{ end }}
+					{{ end }}
+			
+					{{ if eq $server.Hostname "_" }}
+					# health checks in cloud providers require the use of port {{ $all.ListenPorts.HTTP }}
+					location {{ $all.HealthzURI }} {
+						{{ if $all.Cfg.EnableOpentracing }}
+						opentracing off;
+						{{ end }}
+			
+						access_log off;
+						return 200;
+					}
+			
+					# this is required to avoid error if nginx is being monitored
+					# with an external software (like sysdig)
+					location /nginx_status {
+						{{ if $all.Cfg.EnableOpentracing }}
+						opentracing off;
+						{{ end }}
+			
+						{{ range $v := $all.NginxStatusIpv4Whitelist }}
+						allow {{ $v }};
+						{{ end }}
+						{{ if $all.IsIPV6Enabled -}}
+						{{ range $v := $all.NginxStatusIpv6Whitelist }}
+						allow {{ $v }};
+						{{ end }}
+						{{ end -}}
+						deny all;
+			
+						access_log off;
+						stub_status on;
+					}
+			
+					{{ end }}
+			
+			{{ end }}				
+				`,
```

Comment:
For the sake of conciseness can you strip out all the unnecessary parts of the template and test with the minimum possible template instead?